src/tests/krate.rs

```diff
@@ -1178,64 +1178,50 @@ fn summary_new_crates() {
+    let (app, anon, user) = TestApp::with_user();
+    let user = user.as_model();
+
+    app.db(|conn| {
         CrateBuilder::new("foo_download", user.id)
             .version(VersionBuilder::new("1.0.0"))
             .expect_build(&conn);
-    }
-    let resp = t!(middle.call(&mut req));
-    assert_eq!(resp.status.0, 302);
-
-    req.with_path("/api/v1/crates/foo_download/1.0.0/downloads");
-    let mut resp = ok_resp!(middle.call(&mut req));
-    let downloads = ::json::<Downloads>(&mut resp);
-    assert_eq!(downloads.version_downloads.len(), 1);
-    req.with_path("/api/v1/crates/foo_download/downloads");
-    let mut resp = ok_resp!(middle.call(&mut req));
-    let downloads = ::json::<Downloads>(&mut resp);
-    assert_eq!(downloads.version_downloads.len(), 1);
-
-    req.with_path("/api/v1/crates/FOO_DOWNLOAD/1.0.0/download");
-    let resp = t!(middle.call(&mut req));
-    assert_eq!(resp.status.0, 302);
-
-    req.with_path("/api/v1/crates/FOO_DOWNLOAD/1.0.0/downloads");
-    let mut resp = ok_resp!(middle.call(&mut req));
-    let downloads = ::json::<Downloads>(&mut resp);
-    assert_eq!(downloads.version_downloads.len(), 1);
-    req.with_path("/api/v1/crates/FOO_DOWNLOAD/downloads");
-    let mut resp = ok_resp!(middle.call(&mut req));
-    let downloads = ::json::<Downloads>(&mut resp);
-    assert_eq!(downloads.version_downloads.len(), 1);
-
-    let yesterday = Utc::today() + Duration::days(-1);
-    req.with_path("/api/v1/crates/FOO_DOWNLOAD/1.0.0/downloads");
-    req.with_query(&format!("before_date={}", yesterday.format("%F")));
-    let mut resp = ok_resp!(middle.call(&mut req));
-    let downloads = ::json::<Downloads>(&mut resp);
-    assert_eq!(downloads.version_downloads.len(), 0);
-    req.with_path("/api/v1/crates/FOO_DOWNLOAD/downloads");
-    req.with_query(&format!("before_date={}", yesterday.format("%F")));
-    let mut resp = ok_resp!(middle.call(&mut req));
-    let downloads = ::json::<Downloads>(&mut resp);
+    });
+
+    let assert_dl_count = |name_and_version: &str, query: Option<&str>, count: usize| {
+        let url = format!("/api/v1/crates/{}/downloads", name_and_version);
+        let downloads: Downloads = if let Some(query) = query {
+            anon.get_with_query(&url, query).good()
+        } else {
+            anon.get(&url).good()
+        };
+        assert_eq!(downloads.version_downloads.len(), count);
+    };
+
+    let download = |name_and_version: &str| {
+        let url = format!("/api/v1/crates/{}/download", name_and_version);
+        anon.get::<()>(&url).assert_status(302);
+        // TODO: test the with_json code path
+    };
+
+    download("foo_download/1.0.0");
+    assert_dl_count("foo_download/1.0.0", None, 1);
+    assert_dl_count("foo_download", None, 1);
+
+    download("FOO_DOWNLOAD/1.0.0");
+    assert_dl_count("FOO_DOWNLOAD/1.0.0", None, 1);
+    assert_dl_count("FOO_DOWNLOAD", None, 1);
+    // TODO: Is the behavior above a bug?
```

Comment:
shit maybe?? if so it has been since here: https://github.com/rust-lang/crates.io/commit/18c5527968b7cb0b5fc0a9822b85ff6cf5919376